src/librustc/mir/repr.rs

```diff
@@ -334,6 +335,14 @@ pub enum TerminatorKind<'tcx> {
         unwind: Option<BasicBlock>
     },
 
+    /// Drop the Lvalue and write an operand over it
+    DropAndReplace {
+        location: Lvalue<'tcx>,
+        value: Operand<'tcx>,
```

Comment:
For consistency, using `value` and `replacement` might be a better idea here. Having `value` mean different things for `Drop` and `DropAndReplace` seems confusing.